src/FluentAssertions.AspNetCore.Mvc/PhysicalFileResultAssertions.cs

```diff
@@ -1,62 +1,185 @@
-using System;
+﻿using System;
 using System.Diagnostics;
+using FluentAssertions.Execution;
+using FluentAssertions.Primitives;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Net.Http.Headers;
 
 namespace FluentAssertions.AspNetCore.Mvc
 {
     /// <summary>
-    ///     Contains a number of methods to assert that a <see cref="PhysicalFileResult" /> is in the expected state.
+    /// Contains a number of methods to assert that a <see cref="PhysicalFileResult"/> is in the expected state.
     /// </summary>
     [DebuggerNonUserCode]
-    public class PhysicalFileResultAssertions : FileResultAssertions
+    public class PhysicalFileResultAssertions : ObjectAssertions
     {
         #region Public Constructors
 
-        public PhysicalFileResultAssertions(PhysicalFileResult fileResult)
-            : base(fileResult)
+        /// <summary>
+        ///     Initializes a new instance of the <see cref="T:PhysicalFileResultAssertions" /> class.
+        /// </summary>
+        /// <param name="subject">The object to test assertion on</param>
+        public PhysicalFileResultAssertions(PhysicalFileResult subject) : base(subject)
         {
         }
 
         #endregion
 
         #region Public Properties
-
         /// <summary>
-        ///     The <see cref="PhysicalFileResult.FileName">FileName</see> on the <see cref="PhysicalFileResult"/>
+        ///     The value on the PhysicalFileResult
         /// </summary>
         public string FileName => PhysicalFileResultSubject.FileName;
+        public string ContentType => PhysicalFileResultSubject.ContentType;
+        public string FileDownloadName => PhysicalFileResultSubject.FileDownloadName;
+        public DateTimeOffset? LastModified => PhysicalFileResultSubject.LastModified;
+        public EntityTagHeaderValue EntityTag => PhysicalFileResultSubject.EntityTag;
 
-        #endregion Private Properties
+        #endregion
 
         #region Private Properties
-
         private PhysicalFileResult PhysicalFileResultSubject => (PhysicalFileResult)Subject;
 
-        #endregion Private Properties
+        #endregion
 
         #region Public Methods
+        /// <summary>
+        /// Asserts that the FileName is exactly the same as the expected FileName, ignoring the casing.
+        /// </summary>
+        /// <param name="expectedFileName">The expected FileName string.</param>
+        /// <param name="reason">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion 
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="reasonArgs">
+        /// Zero or more objects to format using the placeholders in <see cref="reason" />.
+        /// </param>
+        public PhysicalFileResultAssertions WithPhysicalPath(string expectedFileName, string reason = "", params object[] reasonArgs)
```

Comment:
I created a method WithFileName since in PhysicalFileResult it is also called that. Altought this name sounds reasonable since it is a PhysicalPath I think it is a little confusing there is no such property on the tested class. I am interested in your reasoning.

Reply:
That is absolutely correct.
Unfortuately the Framework is not consistent here.
The names of the parameter of the Result Method and the property of the class are different.
I dediced to use the name of the Methods parameter, because this is what the developer sees. He has furthermore the possibility to test against the Filename property, because it is available as a public property in the Assertion class.

Reply:
Could you show where is the psyhicalPath parameter? I checked the constructor of the PhysicalFileResult and it is fileName also.

Reply:
I am using this library:
Assembly: Microsoft.AspNetCore.Mvc.Core, Version=2.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60

There are 4 overloads of PhysicalFile:
- PhysicalFile(physicalPath, contentType);
- PhysicalFile(physicalPath, contentType, fileDownloadName)
- PhysicalFile(physicalPath, contentType, lastModified, entityTag)
- PhysicalFile(physicalPath, contentType, fileDownloadName, lastModified, entityTag)

Here they use physicalPath, whereas in the PhysicalFileResult class constructor they use Filename. The PhysicalFile Method returns PhysicalFileResult and maps the physicalPath Property to Filename.
My idea was to give the developer the possibility to assert for the Method that he is using in the api controller. This is PhysicalFile(...) .  (see Sample project). Therefore he should be able to assert to the parameter he provided in the PhysicalFile Method. This is physicalPath.
Therefore the assertion Methods name is WithPhysicalPath(...).
He can access the Filename Property as well with the Assertion class as it is public.